website/app/validators.php

```diff
@@ -24,7 +24,10 @@
 }, _('This language is not supported'));
 
 $validator->addValidator('ciphered_password', function ($field, $input, $param = null) {
-    return substr($input[$field], 0, 7) === '$2a$11$';
+    $expectedPasswordPrefix = '$2a$'.sprintf("%02d", GK_PASSWORD_HASH_ROTATION).'$';
+    $expectedMinimalHashRotation = GK_PASSWORD_HASH_ROTATION >= 8;
+    $expectedPasswordCheck = substr($input[$field], 0, 7) === $expectedPasswordPrefix;
```

Comment:
What if `GK_PASSWORD_HASH_ROTATION` is set to really high value such as `123`? Should we care about that case?

Reply:
problem also arise if `GK_PASSWORD_HASH_ROTATION` is < 10

Reply:
I added min an max constraints

Reply:
As such, we prevent nothing as `GK_PASSWORD_MAX_HASH_ROTATION` and `GK_PASSWORD_MIN_HASH_ROTATION` are overridable by environment variables, if you decide to keep the length fixed (the `7` in `substr), then make them just plain constants. `define('GK_PASSWORD_MIN_HASH_ROTATION', 10);`

And then, as it's not a _regular user_ choice to choose the `hash rotation count`, but an admin feature at application startup, why to move the check directly to the config class? And refuse to start if values are out of range.

Reply:
Good point, effectively the health check should not return a valid status if we are out of constraints. It will prevent containers rolling upgrade, which is good.

Reply:
If the health check can return some json structure with the valid/invalid checks that would be awesome.

Reply:
(Sorry to be verbose), then the health check method should be moved in it's own controller class.

Reply:
sure, I'll see that point in my next session if you didnt work on it since